activerecord/lib/active_record/autosave_association.rb

```diff
@@ -371,6 +372,11 @@ def before_save_collection_association
         true
       end
 
+      def after_save_collection_association
+        @new_record_before_save = false
+        true
```

Comment:
We're doing it 5 lines above.  It may no longer be necessary but `autosave_association.rb` is littered with them and removing them is beyond the scope of this ticket.  I'd prefer to just be consistent for now.